### PR TITLE
Kaustavha/needle mods

### PR DESCRIPTION
**Pull request by kaustavha**

### CODE REVIEW

**Comment by kaustavha** [comment]
Should close #76 and #44

File: lib/util/knife.js [hunk 1]
```diff
@@ -22,13 +22,21 @@ var async = require('async');
 var misc = require('./misc');
 var sprintf = require('./sprintf');
 
-
 function knife(args, options) {
   return function(callback) {
-    var env = misc.copyEnv();
-    var knife_path = '/usr/bin/knife';
+    var env = misc.copyEnv(),
+        knife_path_yum = '/usr/local/bin/knife',
```
Comment (mburns) [inline lib/util/knife.js]: nice catch.

File: lib/util/knife.js [hunk 2]
```diff
@@ -75,7 +83,7 @@ function knife(args, options) {
 
 
 function knifeSearch(index, query, options, callback) {
-  if (!callback) {
+  if (typeof callback !== 'function') {
```
Comment (mburns) [inline lib/util/knife.js]: Not sure why this change is required.

Comment (kaustavha) [inline lib/util/knife.js]: This will cause a runtime error if callback isn't defined, e.g.

> if (!callback) {console.log('a');}
ReferenceError: callback is not defined
    at repl:1:6
    at REPLServer.self.eval (repl.js:110:21)
    at repl.js:249:20
    at REPLServer.self.eval (repl.js:122:7)
    at Interface.<anonymous> (repl.js:239:12)
    at Interface.EventEmitter.emit (events.js:95:17)
    at Interface._onLine (readline.js:202:10)
    at Interface._line (readline.js:531:8)
    at Interface._ttyWrite (readline.js:760:14)
    at ReadStream.onkeypress (readline.js:99:10)
Needs to be 
if (typeof callback == 'undefined') {

**Comment by mburns** [comment]
WFM. :+1:
